File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/GetSubmodelsResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.SubmodelDescriptor;

import java.util.ArrayList;
import java.util.List;

public class GetSubmodelsResponse {
    private final List<SubmodelDescriptor> submodels;

    public GetSubmodelsResponse()
    {
        this.submodels = new ArrayList<>();
    }

    public void add(SubmodelDescriptor descriptor)
    {
        submodels.add(descriptor);
    }

    public List<SubmodelDescriptor> getSubmodels() {
        return submodels;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/GetSubmodelsResponse.ts
import { SubmodelDescriptor } from "../../../types/digitaltwin/SubmodelDescriptor";

export class GetSubmodelsResponse {
    private readonly _submodels: Array<SubmodelDescriptor>;

    constructor() {
        this._submodels = [];
    }

    add(descriptor: SubmodelDescriptor): void {
        this._submodels.push(descriptor);
    }

    get submodels(): Array<SubmodelDescriptor> {
        return this._submodels;
    }
}